core/node/app_registry/types/app_metadata.go

```diff
@@ -160,8 +176,10 @@ func ValidateAppMetadata(metadata *protocol.AppMetadata) error {
 		return base.RiverError(protocol.Err_INVALID_ARGUMENT, "metadata is required")
 	}
 
-	if metadata.GetUsername() == "" {
-		return base.RiverError(protocol.Err_INVALID_ARGUMENT, "metadata username is required")
+	// Validate username format
+	if err := ValidateBotUsername(metadata.GetUsername()); err != nil {
+		return base.RiverErrorWithBase(protocol.Err_INVALID_ARGUMENT, "metadata username validation failed", err).
```

Comment:
Because the `base` package is such a common package that we use throughout the code base its usually imported with a dot so it won't require prefixing its logic with `base.` Same for the protocol package.